### PR TITLE
Update user profile picture and save profile detailes.

### LINKED ISSUE #6: Update user profile picture
- [x] Users should be able to see their current profile picture at the page. If they didn't upload the picture yet, then they should see a placeholder.
- [x] Users should be able to update their profile picture.
- [x] Users should not be able to upload images which size is more than 1024x1024px

### PR DESCRIPTION
**Pull request by Washubrnfg85**
Profile-detais-page.html bag fixed. Image uploades, profile detailes saves. Pics folger placed in working directory.

### CODE REVIEW

File: src/main/java/org/linksharing/server/http/ApiRestController.java [hunk 1]
```diff
@@ -69,18 +73,38 @@ ResponseEntity<UserProfileDetails> updateProfileDetails(Principal user,
...
     @PostMapping("/profile/picture")
-    ResponseEntity<?> updateProfilePicture(Principal user) {
-        return null;
+    ResponseEntity<UserProfileDetails> updateProfilePicture(Principal user, @RequestParam("file") MultipartFile file) throws IOException {
+        UserProfileDetails userProfile = profileRepository.findByEmail(user.getName());
+
+        String[] fileNameParts = file.getOriginalFilename().split("\\.");
+        String fileExtension = "." + fileNameParts[fileNameParts.length - 1];
+
+        Path newFileName = Paths.get("D:\\Java_Projects\\link-sharing-app\\pics", userProfile.getEmail() + fileExtension);
```
Comment (eugenesmkn) [inline src/main/java/org/linksharing/server/http/ApiRestController.java]: Does this folder exist on my local machine or a server for deployment?

File: src/main/java/org/linksharing/server/http/ApiRestController.java [hunk 2]
```diff
@@ -69,18 +73,38 @@ ResponseEntity<UserProfileDetails> updateProfileDetails(Principal user,
...
+    ResponseEntity<?> getProfilePicture(Principal user) throws IOException {
 
-        File img = new File("../image.jpg");
+        UserProfileDetails userProfile = profileRepository.findByEmail(user.getName());
+        String imageUrl;
+
+        if (userProfile.getImageUrl() != null) {
+            imageUrl = userProfile.getImageUrl();
+        } else {
+            imageUrl = "src/main/resources/static/placeholder.jpg";
```
Comment (eugenesmkn) [inline src/main/java/org/linksharing/server/http/ApiRestController.java]: https://github.com/eugenesimakin/link-sharing-app/pull/14#discussion_r1436044526

**Review by eugenesimakin — CHANGES_REQUESTED** [review]

File: src/main/java/org/linksharing/server/http/ApiRestController.java (hunk 1 above)
Comment (Washubrnfg85) [reply src/main/java/org/linksharing/server/http/ApiRestController.java]: No, it's not.
So the choosen place is correct, but path should be relative insted of absolute?

Comment (eugenesmkn) [reply src/main/java/org/linksharing/server/http/ApiRestController.java]: Yes.

Did you create the `pics` directory starting the app?

File: src/main/java/org/linksharing/server/http/ApiRestController.java (hunk 2 above)
Comment (Washubrnfg85) [reply src/main/java/org/linksharing/server/http/ApiRestController.java]: Oh, crap)
This path should be relative also. I got that.

**Review by eugenesimakin — APPROVED** [review]

**Pull request merged**